### PR TITLE
show media quality info now playing

### PR DESCRIPTION
**Pull request by whatdoineed2do**
With recent `player` changes, the server plays audio in their native formats / not locked at 44.1/16/2 and some users will want to know the input src being played.

This PR adds source `bit_rate` and input codec (ie mp3/flac/etc) to `media_quality` - this data is exposed in `player_status` which is displayed on `now playing` screen above the scrub head.

media quality

### CODE REVIEW

File: src/transcode.h [hunk 1]
```diff
@@ -144,6 +144,14 @@ transcode_decode_query(struct decode_ctx *ctx, const char *query);
 int
 transcode_encode_query(struct encode_ctx *ctx, const char *query);
 
+/* string representation of the codec
+ *
+ * @in  ctx        decode context
+ * @return         static char* - never NULL, caller does NOT own
```
Comment (ejurgensen) [inline src/transcode.h]: You don't need to specify ownership for const char*. Also please correct "string representation" to "String representation" (or maybe just "Name of the codec"?)

**Comment by whatdoineed2do** [comment]
all resolved, thanks

File: src/player.c [hunk 2]
```diff
@@ -1678,23 +1678,21 @@ get_status(void *arg, int *retval)
...
-      status->quality.codec           = pb_session.playing_now->quality.codec;
-    }
-
   switch (player_state)
     {
       case PLAY_STOPPED:
 	DPRINTF(E_DBG, L_PLAYER, "Player status: stopped\n");
 
 	status->status  = PLAY_STOPPED;
+	if (pb_session.playing_now)
```
Comment (ejurgensen) [inline src/player.c]: Remove this. If player_state is PLAY_STOPPED then playing_now should always be null (unless there is a bug).

Comment (whatdoineed2do) [reply src/player.c]: With the check removed the server SEGVs on startup with the web interface connecting.

Comment (ejurgensen) [reply src/player.c]: I meant remove the entire if clause, i.e. leave case PLAY_STOPPED as it was before

**Comment by chme** [comment]
If you have not yet noticed, the last commit leads to a compile error with the current supported ffmpeg version (https://travis-ci.org/ejurgensen/forked-daapd/jobs/528951835):

```
transcode.c:733:79: error: 'AVFormatContext {aka struct AVFormatContext}' has no member named 'url'
 TF(E_LOG, L_XCODE, "No stream data or decoder for '%s'\n", ctx->ifmt_ctx->url);
```

Apart from that, the media quality exposed is the media quality of the source and not the media quality you are listening to. In my case (listening over airplay and not over local audio), i would get different quality information in the now playing screen (depending on the source), but i am still listening to 44.100 stereo.

A better place for displaying the source media quality would be the details dialog of a track. At least some information is already stored in the db (bitrate, samplerate).

To display the media quality for the now playing track, this needs to be done per output/speaker in order to show the correct values.

**Comment by whatdoineed2do** [comment]
> compile error
Thanks - i had not noticed this; my dev box is running ffmpeg 4.x rather than the 3.x that is on the debian base image.  Wrapped/fixed but you guys may decide to leave as is.

> the media quality exposed is the media quality of the source and not the media quality you are listening to. 

Yes it is the source/original media quality that I want to display; the fact that the output _may_ be downsampling etc is not what I was intending to show.

But a Q; with the recent master's `player` merge, the bitrate is not locked to 44100/2/16 anymore and its the native bitrate (https://github.com/ejurgensen/forked-daapd/issues/395#issuecomment-481866011).  Are we saying that even with this recent change some of subsystems do other transcoding or we saying that the actual h/w data srcs are downsampling - for the latter, we're able to determine this?

But again, it's the original src material bitrate that I was intending to display on the play screen - having it on the modal doesnt achieve the original goal.

**Comment by chme** [comment]
The airplay protocol requires a fixed bitrate of 44100/2/16 (at least version 1 that forked-daapd supports). forked-daapd therefor transcodes the music for all airplay speakers. I guess there is also transcoding involved for chromecast, the http stream and pipe output (but i have not checked).

The bitrate of the file (original source) is fixed and does not change if the song is playing or not. The source for this information should therefor be the files db (if there is an information missing, that could be added during the scan). Getting the value from the player does not make sense in this case.

Prominently displaying this information on the now playing page will be confusing, as it is not the music quality that you are listening to. The available space is limited on this page and i suspect the information for most users is not that important to display that prominently.

It would be fine to add it to the details dialog.

**Pull request closed without merging**